pkg/archive/diff.go

```diff
@@ -173,10 +173,24 @@ func UnpackLayer(dest string, layer ArchiveReader) (size int64, err error) {
 	return size, nil
 }
 
-// ApplyLayer parses a diff in the standard layer format from `layer`, and
-// applies it to the directory `dest`. Returns the size in bytes of the
-// contents of the layer.
+// ApplyLayer parses a diff in the standard layer format from `layer`,
+// and applies it to the directory `dest`. The stream `layer` can be
+// compressed or uncompressed.
```

Comment:
@dmcgowan almost (https://github.com/docker/docker/blob/master/pkg/archive/archive.go#L115) it's after having peeked 10 bytes to tell what kind of compression it is.